packages/charts/src/components/legend/series/series.component.tsx

```diff
@@ -39,7 +41,7 @@ export const SeriesLegend: FC<Props> = ({
         typography={typography}
         markColor={color}
         onClick={(disabled: boolean) => onClick(name, disabled, idx)}
-        text={name}
+        text={Array.isArray(name) ? OTHERS_DATA_KEY : name}
```

Comment:
The legend should not be aware of `pie / donut` context so `OTHERS_DATA_KEY ` needs to be removed from here.